ntoskrnl/ex/sysinfo.c

```diff
@@ -239,6 +245,65 @@ ExLockUserBuffer(
     return STATUS_SUCCESS;
 }
 
+VOID
+NTAPI
+ExGetRawSMBiosTable(
+    PVOID pBuffer,
+    ULONG * ulSize)
+{
+    NTSTATUS Status;
+    PVOID DataBlockObject;
+    PWNODE_ALL_DATA AllData;
+    ULONG BufferSize;
+
+    ASSERT(ulSize != NULL);
+    *ulSize = 0;
+
+    /* Open the data block object for the SMBIOS table */
+    Status = IoWMIOpenBlock(&MSSmBios_RawSMBiosTables_GUID,
+                            WMIGUID_QUERY,
+                            &DataBlockObject);
+    if (!NT_SUCCESS(Status))
+    {
+        DPRINT1("IoWMIOpenBlock failed: 0x%08lx\n", Status);
+        return;
+    }
+
+    /* Query the required buffer size */
+    BufferSize = 0;
+    Status = IoWMIQueryAllData(DataBlockObject, &BufferSize, NULL);
+    if (!NT_SUCCESS(Status))
+    {
+        DPRINT1("IoWMIOpenBlock failed: 0x%08lx\n", Status);
+        return;
+    }
+    
+    AllData = ExAllocatePoolWithTag(PagedPool, BufferSize, 'itfS');
```

Comment:
Where does this pooltag come from?

Reply:
'itfS` => Sfti => System firmware table information

Or I should use existing tags?